birdhouse/pavics-compose.sh

```diff
@@ -105,25 +105,28 @@ if [ x"$1" = x"up" ]; then
 fi
 
 COMPOSE_CONF_LIST="-f docker-compose.yml"
+COMPONENT_OVERRIDES=''
+LOADED_COMPONENTS=''
 for adir in $ALL_CONF_DIRS; do
+  service_name=$(basename "$adir")
   if [ -f "$adir/docker-compose-extra.yml" ]; then
     COMPOSE_CONF_LIST="${COMPOSE_CONF_LIST} -f $adir/docker-compose-extra.yml"
+    LOADED_COMPONENTS="${LOADED_COMPONENTS}\n${service_name}"
   fi
-done
-CONFIGURED_COMPONENTS=''
-for adir in $ALL_CONF_DIRS; do
-  CONFIGURED_COMPONENTS="
-    $CONFIGURED_COMPONENTS
-    $(basename $adir)"
-done
 
-for adir in $ALL_CONF_DIRS; do
+  # If previously loaded components specified overrides for the component that was just loaded, load those overrides now
+  previous_overrides=$(printf '%b' "${COMPONENT_OVERRIDES}" | grep "^$service_name " | sed "s/^${service_name}//g" | tr '\n' ' ')
+  COMPOSE_CONF_LIST="${COMPOSE_CONF_LIST} ${previous_overrides}"
+  # Load overrides for other components unless the component to be overridden hasn't been loaded yet. If the component
+  # hasn't been loaded yet, store a reference to it so that it can be added in as soon as the component is loaded.
   for conf_dir in "$adir"/config/*; do
-    service_name=$(basename "$conf_dir")
+    override_service_name=$(basename "$conf_dir")
     extra_compose="$conf_dir/docker-compose-extra.yml"
     if [ -f "$extra_compose" ]; then
-      if echo "$CONFIGURED_COMPONENTS" | grep -q "$service_name"; then
+      if printf '%b' "${LOADED_COMPONENTS}" | grep -q "^$override_service_name$"; then
```

Comment:
Nice trick `print '%b'`

Reply:
Thanks! I never remember how to use printf so I'm always looking it up